### PR TITLE
Add AV Sync config 

### PR DESCRIPTION
**Pull request by endiirawan**
Add AV sync config in PlaybackParameter for fixing issue where the video & audio is not sync, due to device limitation.

### CODE REVIEW

File: library/core/src/main/java/com/google/android/exoplayer2/audio/SonicAudioProcessor.java [hunk 1]
```diff
@@ -66,6 +66,7 @@
   private int sampleRateHz;
   private float speed;
   private float pitch;
+  private long videoOffsetUs;
```
Comment (ojw28) [inline library/core/src/main/java/com/google/android/exoplayer2/audio/SonicAudioProcessor.java:69]: The changes to this class don't seem to do anything (in particular `videoOffsetUs` doesn't seem to be read by anything? Ditto to the change in `DefaultAudioSink`.

File: library/core/src/main/java/com/google/android/exoplayer2/ExoPlayerImplInternal.java [hunk 2]
```diff
@@ -561,7 +561,7 @@ private void doSomeWork() throws ExoPlaybackException, IOException {
     TraceUtil.beginSection("doSomeWork");
 
     updatePlaybackPositions();
-    long rendererPositionElapsedRealtimeUs = SystemClock.elapsedRealtime() * 1000;
+    long rendererPositionElapsedRealtimeUs = (SystemClock.elapsedRealtime() * 1000) + mediaClock.getPlaybackParameters().videoOffsetUs;
```
Comment (ojw28) [inline library/core/src/main/java/com/google/android/exoplayer2/ExoPlayerImplInternal.java:564]: This isn't really the right place to apply this offset. It should probably be applied somewhere under `DefaultAudioSink.getCurrentPositionUs`, if anywhere.

**Review by ojw28 — COMMENTED** [review]
I'm not sure this is the best way of doing this. Also, it's unclear what the problem is that you're actually trying to solve. Please could you file a bug report [here](https://github.com/google/ExoPlayer/issues/new/choose) that fills in all the sections of the template, so we have a clear idea of what this pull request is actually addressing, what device(s) have this apparently device limitation, and what that limitation is? Thanks.

**Comment by endiirawan** [comment]
@ojw28 this issue is only can be reproduced on the specific device only (which is has closed market in Indonesia). here is the detail issue:

Device          : JellyBean API 17
Device Type : STB
Issue             : 
The audio and video are out of sync (audio is behind the video), the delay between audio and video is constant about 500 ms.   
Not only on Exoplayer, this issue also happen on another player (e.g: NexStreaming, etc)
This issue is happen on this kind of device type only.
This issue was tested happen on all content, both clear & protected content.

By using this workaround, this issue is gone (tested). we can adding the offset minus/plus number to adjust it.  and if we not set the offset value it wont break another part I think.

File: library/core/src/main/java/com/google/android/exoplayer2/audio/SonicAudioProcessor.java (hunk 1 above)
Comment (endiirawan) [reply library/core/src/main/java/com/google/android/exoplayer2/audio/SonicAudioProcessor.java:69]: @ojw28  this is used on this class:
https://github.com/google/ExoPlayer/pull/5766/files#diff-d7c4edd055ad44ffd9ee2b757ac41786

when we set the playbackParameters e.g:
```suggestion
  final long videoOffsetUs = 500 * 1000; // video offset in micro seconds
      PlaybackParameters playbackParameters = new PlaybackParameters(1f, 1f, videoOffsetUs, false);
      player.setPlaybackParameters(playbackParameters);
```

somehow exoplayer will copying that parameters into SonicAudioProcessor:
library/core/src/main/java/com/google/android/exoplayer2/audio/DefaultAudioSink.java **Line:140**
```suggestion
  return new PlaybackParameters(
          sonicAudioProcessor.setSpeed(playbackParameters.speed),	          sonicAudioProcessor.setSpeed(playbackParameters.speed),
          sonicAudioProcessor.setPitch(playbackParameters.pitch),	          sonicAudioProcessor.setPitch(playbackParameters.pitch),
          sonicAudioProcessor.setVideoOffsetUs(playbackParameters.videoOffsetUs),
          playbackParameters.skipSilence);	         
   }
```
so if we dont copy the video offset into SonicAudioProcessor, we will get null when call this:
`mediaClock.getPlaybackParameters().videoOffsetUs`
hence I am adding the video offset on this class too.

File: library/core/src/main/java/com/google/android/exoplayer2/ExoPlayerImplInternal.java (hunk 2 above)
Comment (endiirawan) [reply library/core/src/main/java/com/google/android/exoplayer2/ExoPlayerImplInternal.java:564]: @ojw28  oh i see, it is works too. you are right, this logic is more relevant to be applied into class DefaultAudioSink.
OK I will move this part into  `DefaultAudioSink.getCurrentPositionUs`.

**Comment by ojw28** [comment]
I don't think we really want to take this kind of pull request, for what sounds like a fairly severe issue in the underlying platform, on one very specific device, which will be affecting multiple video players running on that device. It's up to the device manufacturer to fix the underlying problem.

**Pull request closed without merging**